### PR TITLE
Add examples for Nested JSON reader

### PR DESCRIPTION
**Pull request by GregoryKimball**
## Description
Adding some examples to show off the nested type JSON reading

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cudf/cudf/utils/ioutils.py [hunk 1]
```diff
@@ -596,6 +596,32 @@
...
+
+Using the `dtype` argument to specify type casting:
+
+>>> json_str = '{"k1": 1, "k2":[1.5]}'
+>>> cudf.read_json(json_str, engine='cudf_experimental', lines=True, dtype={'k1':float, 'k2':cudf.ListDtype(int)})
+    k1   k2
+0  1.0  [1]
+
+>>>
+
```
Comment (galipremsagar) [inline python/cudf/cudf/utils/ioutils.py:624]:
```suggestion
```

stale code?

**Review by galipremsagar — APPROVED** [review]

**Review by mroeschke — APPROVED** [review]

**Comment by karthikeyann** [comment]
rerun tests

**Comment by galipremsagar** [comment]
rerun tests

**Comment by galipremsagar** [comment]
@gpucibot merge

**Pull request merged**